packages/core/src/consumed-thing.ts

```diff
@@ -82,23 +88,37 @@ export default class ConsumedThing extends TD.Thing implements WoT.ConsumedThing
     }
 
 
-    findForm(forms: Array<TD.Form>, op: string, schemes: string[], idx: number): TD.Form {
+    findForm(forms: Array<TD.Form>, op: string, affordance: Affordance, schemes: string[], idx: number): TD.Form {
         let form = null;
 
         // find right operation and corresponding scheme in the array form
         for (let f of forms) {
-            if (f.op != undefined)
-                if (f.op.indexOf(op) != -1 && f.href.indexOf(schemes[idx] + ":") != -1) {
+            let fop: string | Array<string>;
+            if (f.op != undefined) {
+                fop = f.op;
+            } else {
+                // default "op" values
+                // see https://w3c.github.io/wot-thing-description/#sec-default-values
+                switch (affordance) {
+                    case Affordance.PropertyAffordance:
+                        fop = ["readproperty", "writeproperty"];
+                        break;
+                    case Affordance.ActionAffordance:
+                        fop = "invokeaction";
+                        break;
+                    case Affordance.EventAffordance:
+                        fop = "subscribeevent";
+                        break;
+                }
+            }
+            if (fop != undefined)
```

Comment:
I think this check can be avoided if we initialize fop to an empty string.

Reply:
Sure, fop initialized with an empty string would remove this check.
However, I am not sure if it conveys the same meaning. 
`undefined` to me means *really* undefined while an empty string is actually something else. Am I wrong?

However, either way is fine to me...

Reply:
In general yes `undefined` has a particular meaning, however, in this case, it is used as "default initialization value" so an empty string might work as well. Also, now that I'm looking again at this piece of code, it seems not even possible that `fop` is not assigned to a `not` `undefined` value. In the if branch is assigned to f.op (which is not undefined because of the if boolean expression). On the other hand, in the else branch is assigned to static values, unless we are passing something that is not in the case list (which should throw an Error I guess).

Reply:
I updated the PR accordingly (fop is initialized with empty string)